contracts/SpokePool.sol

```diff
@@ -98,7 +98,11 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         address recipient,
         bool isSlowRelay
     );
-    event RelayedRootBundle(uint32 indexed rootBundleId, bytes32 relayerRefundRoot, bytes32 slowRelayRoot);
+    event RelayedRootBundle(
+        uint32 indexed rootBundleId,
+        bytes32 indexed relayerRefundRoot,
+        bytes32 indexed slowRelayRoot
```

Comment:
Not sure if this is _strictly_ required, but I think it's reasonable to add these as indexed params.

The reason I don't think it's required is that I would expect that we would, for instance, query the last `n` `rootBundleId`s, reconstruct those merkle roots based on finding the corresponding mainnet events based on the associated roots?

Or were you envisioning it starting with a mainnet query that goes back `n` blocks to determine the roots and then finding the associated rootBundleIds on the L2s for those roots?

Not strongly opinionated here, and I think this is reasonable to add regardless, just wanted to better understand the design.

Reply:
> reconstruct those merkle roots based on finding the corresponding mainnet events based on the associated roots?

What do you mean by this?

I was envisioning the dataworker first looking up some `ProposeRootBundle` event on L1. Then it gets the `bundleBlockEvalNumbers` from that event, reconstructs roots for those blocks, and then finds a `RelayedRootBundle` event matching those roots

Reply:
I think the spokepool & hubpool have all this information in their clients already and I’m pretty sure you can do this join without needing these to be indexed with all the data available in the clients you can look at `RelayedRootBundle` events on the spoke pool and relate it to `ProposeRootBundle` events by joining on the `slowRelayRoot` or `relayerRefundRoot` and from this you can find the `rootBundleId`.

i.e:

1. find all `RelayedRootBundle`. 
2. For each ^^, join to an an associated event on the `HubPool` via the `ProposeRootBundle `event.
3. from this, find the associated `bundleEvaluationBlockNumbers`
4. use ^^ to find all valid deposits in period to reconstruct tree.
5. construct each leaf and associated proof. 
6. simulate execution on target chain.
7. If it can be executed, execute.

alternative to 6 and 7 is to look at `ExecutedRelayerRefundRoot` and work out which are still to be executed.

Reply:
> reconstruct those merkle roots based on finding the corresponding mainnet events based on the associated roots?

What do you mean by this?

I was envisioning the dataworker first looking up some `ProposeRootBundle` event on L1. Then it gets the `bundleBlockEvalNumbers` from that event, reconstructs roots for those blocks, and then finds a `RelayedRootBundle` event matching those roots

Reply:
@chrismaree why do you need to look up ALL `RelayedRootBundle` though to execute leaves for 1 bundle block range?

Reply:
> @chrismaree why do you need to look up ALL RelayedRootBundle though to execute leaves for 1 bundle block range?

you dont need to but we have them anyways in the client :)

Reply:
Hmm then I see your point that we don't need this indexed... what to do?

Reply:
Maybe someone else would want to index it since matching roots is the canonical way to identify `rootBundleId`?